File: solutions/python/kindergarten-garden/1/kindergarten_garden.py
class Garden:
    students = [
        "Alice",
        "Bob",
        "Charlie",
        "David",
        "Eve",
        "Fred",
        "Ginny",
        "Harriet",
        "Ileana",
        "Joseph",
        "Kincaid",
        "Larry",
    ]

    _LET_TO_PLANT = {
        "V": "Violets",
        "R": "Radishes",
        "G": "Grass",
        "C": "Clover",
    }

    def __init__(self, diagram, students=None):
        self.students = sorted(students) if students is not None else Garden.students
        self.rows = diagram.split("\n")

    def plants(self, name):
        idx = self.students.index(name)
        return [
            Garden._LET_TO_PLANT.get(row[i])
            for row in self.rows
            for i in range(idx * 2, idx * 2 + 2)
        ]

File: solutions/fsharp/kindergarten-garden/1/KindergartenGarden.fs
module KindergartenGarden

type Plant =
    | Grass
    | Clover
    | Radishes
    | Violets

let plants (diagram: string) (student: string) =
    let idx = int student.[0] - int 'A'

    let map char =
        match char with
        | 'G' -> Grass
        | 'C' -> Clover
        | 'R' -> Radishes
        | _ -> Violets

    let result =
        diagram.Split('\n')
        |> Seq.collect (fun line -> line.[idx * 2 .. idx * 2 + 1] |> Seq.map map)

    result |> List.ofSeq
